sled-storage/src/manager.rs

```diff
@@ -590,9 +663,145 @@ impl StorageManager {
             info!(self.log, "KeyManager ready, but no ledger detected");
         }
 
+        // We don't load any configuration for datasets, since we aren't
+        // currently storing any dataset information in-memory.
+        //
+        // If we ever wanted to do so, however, we could load that information
+        // here.
+
         Ok(())
     }
 
+    async fn datasets_ensure(
+        &mut self,
+        mut config: DatasetsConfig,
+    ) -> Result<DatasetsManagementResult, Error> {
+        let log = self.log.new(o!("request" => "datasets_ensure"));
+
+        // Ensure that the datasets arrive in a consistent order
+        config.datasets.sort_by(|a, b| a.id.partial_cmp(&b.id).unwrap());
```

Comment:
Would it help to use an already-sorted type, like a `BTreeMap` here? Related, I might make a comment on the `DatasetsManagementResult` type that the result entries may be in a different order than the input. We could also use a sorted type there too, if that were preferable.

Reply:
I did not realize progenitor worked with `BTreeMap`! Neat, I'll update the API.